modules/promotion/src/Entity/Promotion.php

```diff
@@ -243,9 +243,23 @@ public function apply(EntityInterface $entity) {
     $entity_type_id = $entity->getEntityTypeId();
     // @todo should whatever invokes this method be providing the context?
     $context = new Context(new ContextDefinition('entity:' . $entity_type_id), $entity);
+
+    // Execute each plugin, this is an AND operation.
+    // @todo support OR operations.
+    /** @var \Drupal\commerce\Plugin\Field\FieldType\PluginItem $item */
+    foreach ($this->get('conditions') as $item) {
```

Comment:
If a promotion applies that doesn't match the rules, that is going to confuse the hell out of any admins though. If a API user wants to apply a promotion outside of conditions, wouldn't it be better for them to make a new promotion? Also any re-validation of the order based on a change would cause it to be dumped? either way it would be confusing.